src/Bindings/PluginManager.cpp

```diff
@@ -223,14 +223,15 @@ void cPluginManager::Tick(float a_Dt)
 
 
 
-bool cPluginManager::CallHookBlockSpread(cWorld & a_World, int a_BlockX, int a_BlockY, int a_BlockZ, eSpreadSource a_Source)
+template <cPluginManager::PluginHook HookName, class ... PmfArgs, class ... RealArgs>
```

Comment:
Do PmfArgs and RealArgs have to be separate? This works for me on cpp.sh:
```
// Example program
#include <iostream>
#include <string>

using namespace std;

template <class ... RealArgs>
bool foo(bool (*fn)(RealArgs...), RealArgs && ... a_Args) {
    return fn(forward<RealArgs>(a_Args)...); // also, std::forward with universal references
}

bool bar(int x) {
    return x > 5;
}

bool bar2(int x, int y) {
    return x > y;
}

int main()
{
    cout << foo(bar, 5) << endl;
    cout << foo(bar2, 5, 4) << endl;
}
```

(for better or worse, I've been learning a lot of C++ lately, now I'm stuck thinking about things like parameter packs and universal references)

Reply:
It doesn't work where the deduced type is different from the function signature. [For example](http://cpp.sh/4ndb) changing the above to:
``` cpp
bool bar(int& x) {
    return x > 5;
}
```

Failes to compile with this error:
```
 In function 'int main()':
29:23: error: no matching function for call to 'foo(bool (&)(int&), int)'
29:23: note: candidate is:
15:6: note: template<class ... RealArgs> bool foo(bool (*)(RealArgs ...), RealArgs&& ...)
15:6: note: template argument deduction/substitution failed:
29:23: note: inconsistent parameter pack deduction with 'int&' and 'int'
```

Reply:
That fails because it's passing bar() a rvalue (I think? Or is it an xvalue?), but bar is expecting a lvalue reference.

If you replace bar's literal parameter with a variable, it compiles fine:
http://cpp.sh/27kbz

Reply:
Compilation fails before it even tries to call bar, the issue is the type deduction.  Perhaps it was a bad example though.

The other way around still fails as well though:
http://cpp.sh/7qkan